panther_manager/include/panther_manager/shutdown_host_node.hpp

```diff
@@ -0,0 +1,109 @@
+#ifndef PANTHER_MANAGER_SHUTDOWN_HOST_NODE_HPP_
+#define PANTHER_MANAGER_SHUTDOWN_HOST_NODE_HPP_
+
+#include <behaviortree_cpp/action_node.h>
+#include <behaviortree_cpp/bt_factory.h>
+#include <libssh/libssh.h>
+
+#include <ros/ros.h>
+
+namespace panther_manager
+{
+
+class ShutdownHost : public BT::StatefulActionNode
+{
+public:
+  ShutdownHost(const std::string & name, const BT::NodeConfig & conf)
+  : BT::StatefulActionNode(name, conf)
+  {
+    node_name_ = ros::this_node::getName();
+  }
+
+  ssh_session session_;
+  ssh_channel channel_;
+
+  bool check_ip(const std::string ip)
+  {
+    return system(("ping -c 1 -w 1 " + ip + " > /dev/null").c_str()) == 0;
+  }
+
+  int ssh_execute_command(const char * host, const char * user, const char * command)
+  {
+    ROS_INFO("[%s] Shutting down device at: %s", node_name_.c_str(), host);
+
+    session_ = ssh_new();
+    if (session_ == NULL) return -1;
+
+    ssh_options_set(session_, SSH_OPTIONS_HOST, host);
+    ssh_options_set(session_, SSH_OPTIONS_USER, user);
+    ssh_options_set(session_, SSH_OPTIONS_PORT, &port_);
+    ssh_options_set(session_, SSH_OPTIONS_LOG_VERBOSITY, &verbosity_);
+
+    int rc = ssh_connect(session_);
+    if (rc != SSH_OK) {
```

Comment:
```suggestion
    if (ssh_connect(session_) != SSH_OK) {
```
Since `rc` is not used later this could be reduced. The same could be applied later on